modules/platforms/python/cpp_module/type_conversion.h

```diff
@@ -122,40 +161,162 @@ static void submit_pyobject(ignite::binary_tuple_builder &builder, PyObject *obj
         return;
     }
 
-    if (PyBool_Check(obj)) {
-        bool val = (obj == Py_True);
+    if (PyObject_IsInstance(obj, py_get_module_timestamp_class())) {
+        auto double_value = PyFloat_AsDouble(obj);
+        if (PyErr_Occurred()) {
+            throw ignite::ignite_error("Can not get ticks count from a TIMESTAMP value");
+        }
+
+        auto secs = std::int64_t(std::trunc(double_value));
+        auto nanos = std::lround((double_value - secs) * 1'000'000'000);
+        ignite::ignite_timestamp value(secs, nanos);
         if (claim) {
-            ignite::protocol::claim_type_and_scale(builder, ignite::ignite_type::BOOLEAN);
-            builder.claim_bool(val);
+            ignite::protocol::claim_type_and_scale(builder, ignite::ignite_type::TIMESTAMP);
+            builder.claim_timestamp(value);
         } else {
-            ignite::protocol::append_type_and_scale(builder, ignite::ignite_type::BOOLEAN);
-            builder.append_bool(val);
+            ignite::protocol::append_type_and_scale(builder, ignite::ignite_type::TIMESTAMP);
+            builder.append_timestamp(value);
         }
         return;
     }
 
-    if (PyBytes_Check(obj)) {
-        auto *data = reinterpret_cast<std::byte*>(PyBytes_AsString(obj));
-        auto len = PyBytes_Size(obj);
-        ignite::bytes_view view(data, len);
+    if (PyObject_IsInstance(obj, py_get_module_number_class())) {
+        auto obj_str = PyObject_Str(obj);
+        if (!obj_str) {
+            throw ignite::ignite_error("Can not convert NUMBER to str");
```

Comment:
Here and other places - if a conversion fails, we should provide more details:
- The value that we could not convert
- The original error from converter, if applicable